Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/MSBuildTemplates/MSBuildForUnity.Common.props.template

```diff
@@ -12,4 +12,8 @@
     <UnityMajorVersion><!--UNITY_MAJOR_VERSION_TOKEN--></UnityMajorVersion>
     <UnityMinorVersion><!--UNITY_MINOR_VERSION_TOKEN--></UnityMinorVersion>
   </PropertyGroup>
+  
+  <Target Name="_RemoveOutputDirectory" AfterTargets="Clean">
```

Comment:
I'm still unclear why this is needed. If I create a default SDK style project and add a package reference to MSBuildForUnity and Newtonsoft.Json, it behaves like I would expect: build results in the project dll plus the Newtonsoft dll in the output directory, and clean removes all files from the output directory.

Reply:
That is not what I am seeing, and there are GitHub issues detailing similar problems:
https://github.com/Microsoft/msbuild/issues/2408

Unless I am misunderstanding something, would be great to see an example to understand what I am doing wrong.

Reply:
I found the issue, and don't know how to fix it. Basically if the OutputPath of the project is under the directory that the project is contained in, then Clean works as expected; otherwise Clean doesn't remove the NuGet package output. 

Opened #71.